lake/data/vector.go

```diff
@@ -18,6 +19,10 @@ import (
 func CreateVector(ctx context.Context, engine storage.Engine, path *storage.URI, id ksuid.KSUID) error {
 	get, err := engine.Get(ctx, SequenceURI(path, id))
 	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			// Make a cleaner error.
+			err = fmt.Errorf("object %s: does not exist", id)
```

Comment:
Nit: Preserve the underlying error so the API stays consistent.
```suggestion
			err = fmt.Errorf("object %s: %w", id, fs.ErrNotExist)
```